scripts/valhalla_build_elevation

```diff
@@ -1,81 +1,249 @@
+    max_x, max_y = [ceil(x) for x in bounds[-2:]]
+    for x in range(min_x, max_x):
+        for y in range(min_y, max_y):
+            yield [x, y, x + 1, y + 1]
+
+
+def get_tiles_with_geojson(input_dir: Path) -> Set[Tile]:
+    try:
+        from shapely.geometry import Polygon, box
+    except ImportError:
+        LOGGER.critical(
+            "Could not import shapely. Please install shapely or use another download method instead."
+        )
+        sys.exit(1)
+
+    if not input_dir.is_dir() or not len(list(input_dir.glob('*.geojson'))) > 0:
+        LOGGER.critical(f"Geojson directory does not exist or contains no GeoJSON files: {input_dir.resolve()}")
+        sys.exit(1)
+
+    def get_outer_rings(input_dir: Path) -> Iterable[Polygon]:
+        for file in input_dir.glob("*.geojson"):
+            with open(file) as f:
+                geojson = json.load(f)
+            for feature in geojson["features"]:
+                if feature["geometry"]["type"] == "Polygon":
+                    yield Polygon(feature["geometry"]["coordinates"][0])
+                if feature["geometry"]["type"] == "MultiPolygon":
+                    for single_polygon in feature["geometry"]["coordinates"]:
+                        yield Polygon(single_polygon[0])
+
+    tile_infos = set()
+    for poly in get_outer_rings(input_dir):
+        for rect in grid_from_bounds(poly.bounds):
+            if poly.intersects(box(*rect)):
+                s, w, *_ = rect
+                tile_infos.add(get_tile_info(s, w))
+
+    return tile_infos
+
+
+def get_tiles_with_bbox(bbox_str: str) -> Set[Tile]:
+    try:
+        bbox = min_x, min_y, max_x, max_y = [float(x) for x in bbox_str.split(",")]
+    except ValueError:
+        LOGGER.critical(f"BBOX {bbox_str} is not a comma-separated string of coordinates.")
+        sys.exit(1)
+
+    # validate bbox
+    if (
+        min_x > max_x
+        or min_x < -180
+        or max_x > 180
+        or min_y > max_y
+        or min_y < -90
+        or max_y > 90
+    ):
+        LOGGER.critical(f"Bbox invalid: {bbox}")
+        sys.exit(1)
+
+    LOGGER.debug(f"Received valid bbox: {bbox}")
+
+    tile_infos = set()
+    for grid in grid_from_bounds(bbox):
+        s, w = grid[:2]
+        tile_infos.add(get_tile_info(s, w))
+
+    return tile_infos
+
+
+def get_tiles_with_graph(graph_dir: Path) -> Set[Tile]:
+    if not graph_dir.is_dir():
+        LOGGER.critical(f"Graph directory {graph_dir.resolve()} does not exist.")
+        sys.exit(1)
+
+    tile_infos = set()
+    local_dir = graph_dir.joinpath('2')
+    tile_width = int(360 / LOCAL_SIZE)
+    for tile_fp in local_dir.rglob('*.gph'):
+        # turn the path into a tile ID
+        tile_id = int(str(tile_fp.parent.relative_to(local_dir).joinpath(tile_fp.stem)).replace(os.sep, ''))
+        s, w = floor(int(tile_id % tile_width) * LOCAL_SIZE - 180), floor(int(tile_id / tile_width) * LOCAL_SIZE - 90)
+
+        tile_infos.add(get_tile_info(s, w))
+
```

Comment:
I think this should be correct now @kevinkreiser : we floor the lower-left corner of local tiles giving us the integer degree which corresponds to the elevation tile notation. it's now a `set` so we don't collect dups.